FRC-Extension/RoboRIOCode/DeployManager.cs

```diff
@@ -39,106 +44,81 @@ public DeployManager(DTE dte)
 
             //Connect to Robot Async
             await OutputWriter.Instance.WriteLineAsync("Attempting to Connect to RoboRIO").ConfigureAwait(false);
-            Task<bool> rioConnectionTask = StartConnectionTaskAsync(teamNumber);
-            Task delayTask = Task.Delay(10000);
+
+
+
+            Task<RoboRioConnection> rioConnectionTask = RoboRioConnection.StartConnectionTaskAsync(teamNumber);
 
             CodeReturnStruct codeReturn = await BuildAndPrepareCodeAsync(debug, robotProject).ConfigureAwait(false);
 
             if (codeReturn == null) return false;
 
             await writer.WriteLineAsync("Waiting for Connection to Finish").ConfigureAwait(false);
-            if (await Task.WhenAny(rioConnectionTask, delayTask).ConfigureAwait(false) == rioConnectionTask)
+            m_roboRioConnection = await rioConnectionTask.ConfigureAwait(false);
```

Comment:
Dispose disposes it. That's why using was added to all usages of DeployManager class. I could make it in a using statement in the function, but then I would have to pass it manually to the rest of the functions later.